draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -848,6 +851,151 @@ Content-Type: text/plain
 </artwork>
 </section>
 
+<section title="Origin" anchor="http.origin">
+<t>
+   The origin for a given URI is the triple of scheme, host, and port
+   after normalizing the scheme and host to lowercase and normalizing the
+   port to remove any leading zeros. If port is elided from the URI, the
+   default port for that scheme is used. For example, the URI
+</t>
+<artwork type="example">
+   https://Example.Com/happy.js
+</artwork>
+<t>
+   would have the origin
+</t>
+<artwork type="example">
+   { "https", "example.com", "443" }
+</artwork>
+<t>
+   which can also be described as the normalized URI prefix with port always
+   present:
+</t>
+<artwork type="example">
+   https://example.com:443
+</artwork>
+<t>
+   The notion of an origin for any given URI is central to the semantics of
+   HTTP. Each origin defines its own namespace and controls how identifiers
+   within that namespace are mapped to resources. In turn, how the origin
+   responds to valid requests, consistently over time, determines the
+   semantics that users will associate with a URI, and the usefulness of
+   those semantics is what ultimately transforms these mechanisms into a
+   "resource" for users to reference and access in the future.
+</t>
+<t>
+   Two origins are distinct if they differ in scheme, host, or port. Even
+   when it can be verified that the same entity controls two distinct origins,
+   the two namespaces under those origins are distinct unless explicitly
+   aliased by a server authoritative for that origin.
+</t>
+<t>
+   Origin is also used within HTML and related Web protocols, beyond the
+   scope of this document, as described in <xref target="RFC6454"/>.
+</t>
+<t>
+   Note that the presence of a URI does not imply that there is always an
+   HTTP server at the identified origin listening for connections on
+   the identified host and port. Anyone can mint a URI. What the origin
+   determines is who has the right to respond authoritatively to requests that
+   target the identified resource. The delegated nature of registered names
+   and IP addresses creates a federated namespace whether or not an HTTP
+   server is present.
+</t>
+<t>
+   Although HTTP is independent of the transport protocol, the "http" scheme
+   is specific to associating authority with whomever controls the origin
+   server listening for TCP connections on the indicated port of whatever
+   host is identified within the authority component. This is a very weak
+   sense of authority because it depends on both client-specific name
+   resolution mechanisms and communication that might not be secured from
+   man-in-the-middle attacks. Nevertheless, it is a sufficient minimum for
+   binding "http" identifiers to an origin server for consistent resolution
+   within a trusted environment.
```

Comment:
I don't like this concept of "trusted environment".  I would rather not defend the "http" scheme at all, simply recognizing the inherent limitations being enough for this document.  Saying how it is should be enough; value judgments can be made by others.

Reply:
It is important to remind people that HTTP is used in a lot of contexts, including things like non-TLS secured networks and localhost sockets. This is the easy way to phrase it.

Reply:
Easy doesn't mean right though.  The challenge here is that trusted environment implies that it is possible to trust an environment.  I would say instead:

> In some contexts, clients might choose to accept the risk of attack, or choose to rely on other mechanisms to mitigate those risks.

What I think this needs to capture is that this risk remains even in those places.  I appreciate that this is a value judgment, but I think that it's an important one.